### PR TITLE
Add unblinded flag , alternative edition

### PR DESCRIPTION
**Pull request by Kukks**

### CODE REVIEW

File: NBXplorer/LiquidRepository.cs [hunk 1]
```diff
@@ -206,6 +208,30 @@ protected override async Task AfterMatch(TrackedTransaction tx, IReadOnlyCollect
 			}
 		}
 
+		
+		public override async Task<KeyPathInformation> GetUnused(DerivationStrategyBase strategy, DerivationFeature derivationFeature, int n, bool reserve)
```
Comment (NicolasDorier) [inline NBXplorer/LiquidRepository.cs]: This hack will probably have side effect which will be worse than just considering blinded and unblinded as different wallet.

I don't understand why you need to do this. No user will want one store in blinded and the other unblinded and share the same indexes.

**Comment by Kukks** [comment]
The issue here is that if you are tracking the same xpub with blinded and
unblinded options, they will generate the same script pub when requesting
an unused address. Once a payment is made to the blinded address, it will
trigger the unblinded tracker too.

If you have two stores on btcpay, each using the same xpub:
Store A: uses xpub - [unblinded
Store B: uses xpub
Generate invoice on both stores.
Both invoices have the same script pub key, but different addresses( store
b is the blinded address format, but its unblinded address is the same)

If you pay Store A s invoice, the invoice from store B is also paid.
If you pay Store B s invoice,the invoice from store A is not paid as it is
blinded

**Comment by NicolasDorier** [comment]
Good point. Though I am not convinced it is a problem worth solving, as it means that a user need:

1. Use lightning
2. Have two stores
3. Configure both stores with same xpub but and one blinded/other unblinded

File: NBXplorer/LiquidRepository.cs (hunk 1 above)
Comment (Kukks) [reply NBXplorer/LiquidRepository.cs]: It's more for the case that a user starts off with one option and then switches to the other, only to find out that there is address reuse happening and that some calls to GetUnused would actually return indexes that have been used. 

Maybe I am overthinking this?

**Comment by Kukks** [comment]
> 
> 
> Good point. Though I am not convinced it is a problem worth solving, as it means that a user need:
> 
>     1. Use lightning
> 
>     2. Have two stores
> 
>     3. Configure both stores with same xpub but and one blinded/other unblinded
> 
> 
> Not really something that should happen everyday...

Why do you need to use lightning?

**Comment by NicolasDorier** [comment]
I really don't like this fix though.
I can't convince myself there is not side effect which will waste our time in debugging later. And it is not easy to understand: Why would different label sometimes map to the same addresses and sometime not?

This is also for fixing something that will happen only one time for a very limited set of people.

**Comment by Kukks** [comment]
Fair enough, will remove the hack. I guess a documented solution would be
to Rescan if you encounter the issue

**Comment by Kukks** [comment]
updated and removed the workaround

File: NBXplorer.Client/DerivationStrategy/IStrategy.cs [hunk 2]
```diff
@@ -38,11 +41,18 @@ public Derivation GetDerivation(KeyPath keyPath)
 		}
 		public abstract Derivation GetDerivation();
 
-		protected abstract string StringValue
+		protected abstract string StringValueCore
```
Comment (NicolasDorier) [inline NBXplorer.Client/DerivationStrategy/IStrategy.cs]: remove this refactoring, this is not needed anymore

File: NBXplorer/Extensions.cs [hunk 3]
```diff
@@ -48,7 +48,7 @@ internal static void AddRange<T>(this HashSet<T> hashset, IEnumerable<T> element
 		}
 		internal static uint160 GetHash(this DerivationStrategyBase derivation)
 		{
-			var data = Encoding.UTF8.GetBytes(derivation.ToString());
+			var data = Encoding.UTF8.GetBytes(derivation.StringValue);
```
Comment (NicolasDorier) [inline NBXplorer/Extensions.cs]: remove this

File: NBXplorer.Client/DerivationStrategy/IStrategy.cs (hunk 2 above)
Comment (Kukks) [reply NBXplorer.Client/DerivationStrategy/IStrategy.cs]: This one is needed in order to not make each derivation type have to do `StringValue => $"{StringValueCore}{GetSuffixOptionsString()}";`

**Comment by NicolasDorier** [comment]
You need to add tests to make sure the that the derivation schemes are insensitive to the order and the casing of the labels. Else we will end up with various issues.

**Comment by Kukks** [comment]
> 
> 
> You need to add tests to make sure the that the derivation schemes are insensitive to the order and the casing of the labels. Else we will end up with various issues. (only need tests at the factory level)

Done

**Pull request merged**